metals/src/main/scala/scala/meta/internal/metals/CodeActionProvider.scala

```diff
@@ -2,27 +2,56 @@ package scala.meta.internal.metals
 
 import org.eclipse.{lsp4j => l}
 import scala.meta.pc.CancelToken
+import scala.meta.internal.mtags.Semanticdbs
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
+import scala.collection.JavaConverters._
 
 final class CodeActionProvider(
-    compilers: Compilers
+    compilers: Compilers,
+    trees: Trees,
+    buffers: Buffers,
+    semanticdbs: Semanticdbs,
+    symbolSearch: MetalsSymbolSearch,
+    definitionProvider: DefinitionProvider
 ) {
 
   def codeActions(
       params: l.CodeActionParams,
       token: CancelToken
   )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = {
 
-    for {
-      quickFixes <- Future
-        .sequence(
-          List(QuickFix.ImportMissingSymbol)
-            .map(_.contribute(params, compilers, token))
-        )
-        .map(_.flatten)
-    } yield quickFixes
+    val allActions = List(
+      QuickFixes.ImportMissingSymbol,
+      Refactorings.UseNamedArguments
+    )
+
+    val isRequestedKind: CodeAction => Boolean =
```

Comment:
nit: can we refactor this into `def isRequestedKind(..): Boolean` method?

Reply:
alternatively, inline this into `filter(..)`

Reply:
Done